From stern@rowland.harvard.edu Thu Oct  6 13:46:21 2005
Date: Thu, 6 Oct 2005 16:43:44 -0400 (EDT)
From: Alan Stern <stern@rowland.harvard.edu>
To: Vojtech Pavlik <vojtech@suse.cz>
cc: Greg KH <greg@kroah.com>
Subject: [PATCH] USB: Remove erroneous disconnection tests from hid-core.c
Message-ID: <Pine.LNX.4.44L0.0510061639190.4516-100000@iolanthe.rowland.org>

This patch (as576) removes some mistaken tests for disconnection from
the HID driver.  -EILSEQ refers to an arbitrary low-level protocol
error, not necessarily a disconnection.  Also, a completion routine will
never see a status of -EPERM; that's used only to indicate a failure
during usb_submit_urb.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/hid-core.c |    4 ----
 1 file changed, 4 deletions(-)

--- gregkh-2.6.orig/drivers/usb/input/hid-core.c
+++ gregkh-2.6/drivers/usb/input/hid-core.c
@@ -923,9 +923,7 @@ static void hid_irq_in(struct urb *urb, 
 			break;
 		case -ECONNRESET:	/* unlink */
 		case -ENOENT:
-		case -EPERM:
 		case -ESHUTDOWN:	/* unplug */
-		case -EILSEQ:		/* unplug timeout on uhci */
 			return;
 		case -ETIMEDOUT:	/* NAK */
 			break;
@@ -1099,7 +1097,6 @@ static void hid_irq_out(struct urb *urb,
 		case 0:			/* success */
 			break;
 		case -ESHUTDOWN:	/* unplug */
-		case -EILSEQ:		/* unplug timeout on uhci */
 			unplug = 1;
 		case -ECONNRESET:	/* unlink */
 		case -ENOENT:
@@ -1147,7 +1144,6 @@ static void hid_ctrl(struct urb *urb, st
 				hid_input_report(hid->ctrl[hid->ctrltail].report->type, urb, 0, regs);
 			break;
 		case -ESHUTDOWN:	/* unplug */
-		case -EILSEQ:		/* unplug timectrl on uhci */
 			unplug = 1;
 		case -ECONNRESET:	/* unlink */
 		case -ENOENT:
